uPortSDK/Classes/Crypto/Crypto.swift

```diff
@@ -0,0 +1,123 @@
+//
+//  Crypto.swift
+//  uPortSDK
+//
+
+import Foundation
+import Sodium
+
+/**
+ * This struct exposes methods to encrypt and decrypt messages according to the uPort spec at
+ * https://github.com/uport-project/specs/blob/develop/messages/encryption.md
+ */
+public struct Crypto {
+    private static let ASYNC_ENC_ALGORITHM = "x25519-xsalsa20-poly1305"
+    public static let BLOCK_SIZE = 64
+    
+    /**
+     * This class encapsulates an encrypted message that was produced using
+     * https://github.com/uport-project/specs/blob/develop/messages/encryption.md
+     */
+    public struct EncryptedMessage: Codable {
+        var cipherText: String
+        var nonce: String
+        var ephemPublicKey: String
+        var version: String = ASYNC_ENC_ALGORITHM
+        
+        public init(cipherText: String, nonce: String, ephemPublicKey: String) {
+            self.cipherText = cipherText
+            self.nonce = nonce
+            self.ephemPublicKey = ephemPublicKey
+        }
+        
+        public func encode() -> Data {
+            return try! Data(JSONEncoder().encode(self))
+        }
+        
+        public static func decode(jsonData: Data) -> EncryptedMessage {
+            return try! JSONDecoder().decode(EncryptedMessage.self, from: jsonData)
+        }
+    }
+    
+    /**
+     Encrypts a message with a sender's public key, recipient's secret key, and encryption nonce.
```

Comment:
Slight typo in the comment. The keys are opposite, the receiver public key and sender secret key.